lib/codecs/src/decoding/format/avro.rs

```diff
@@ -0,0 +1,366 @@
+pub struct AvroDeserializer {
+    schema: apache_avro::Schema,
+    strip_schema_id_prefix: bool,
+}
+
+impl AvroDeserializer {
+    /// Creates a new `AvroDeserializer`.
+    pub const fn new(schema: apache_avro::Schema, strip_schema_id_prefix: bool) -> Self {
+        Self {
+            schema,
+            strip_schema_id_prefix,
+        }
+    }
+}
+
+impl Deserializer for AvroDeserializer {
+    fn parse(
+        &self,
+        bytes: Bytes,
+        log_namespace: LogNamespace,
+    ) -> vector_common::Result<SmallVec<[Event; 1]>> {
+        // Avro has a `null` type which indicates no value.
+        if bytes.is_empty() {
+            return Ok(smallvec![]);
+        }
+
+        let bytes = if self.strip_schema_id_prefix {
+            if bytes.len() >= CONFLUENT_SCHEMA_PREFIX_LEN && bytes[0] == CONFLUENT_MAGIC_BYTE {
+                bytes.slice(CONFLUENT_SCHEMA_PREFIX_LEN..)
+            } else {
+                return Err(vector_common::Error::from(
+                    "Expected avro datum to be prefixed with schema id",
+                ));
+            }
+        } else {
+            bytes
+        };
+
+        let value = apache_avro::from_avro_datum(&self.schema, &mut bytes.reader(), None)?;
+
+        let apache_avro::types::Value::Record(fields) = value else {
+            return Err(vector_common::Error::from("Expected an avro Record"));
+        };
+
+        let mut log = LogEvent::default();
+        for (k, v) in fields {
+            log.insert(event_path!(k.as_str()), try_from(v)?);
+        }
+
+        let mut event = Event::Log(log);
+        let event = match log_namespace {
+            LogNamespace::Vector => event,
+            LogNamespace::Legacy => {
+                if let Some(timestamp_key) = log_schema().timestamp_key_target_path() {
+                    let log = event.as_mut_log();
+                    if !log.contains(timestamp_key) {
+                        let timestamp = Utc::now();
+                        log.insert(timestamp_key, timestamp);
+                    }
+                }
+                event
+            }
+        };
+        Ok(smallvec![event])
+    }
+}
+
+// Can't use std::convert::TryFrom because of orphan rules
+pub fn try_from(value: AvroValue) -> vector_common::Result<VrlValue> {
+    // Very similar to avro to json see `impl std::convert::TryFrom<AvroValue> for serde_json::Value`
+    // LogEvent has native support for bytes, so it is used for Bytes and Fixed
+    match value {
+        AvroValue::Array(array) => {
+            let mut vector = Vec::new();
+            for item in array {
+                vector.push(try_from(item)?);
+            }
+            Ok(VrlValue::Array(vector))
+        }
+        AvroValue::Boolean(boolean) => Ok(VrlValue::from(boolean)),
+        AvroValue::Bytes(bytes) => Ok(VrlValue::from(bytes)),
+        AvroValue::Date(d) => Ok(VrlValue::from(d)),
+        AvroValue::Decimal(ref d) => Ok(<Vec<u8>>::try_from(d)
```

Comment:
Let's also return an `Err` here for now.

In case you want to dig into this deeper:

An easier way to convert to bytes:
```
let decimal_bytes: Vec<u8> = d.try_into()?;
Ok(VrlValue::from(decimal_bytes))
```
            
The roundtrip for this fails at the encoder level, specifically when resolving the schema.  I guess problem has to do with specifying `"logicalType":"decimal"` in `decimal_var.avsc` but the bytes test passes.